internal/key/store.go

```diff
@@ -2,11 +2,19 @@ package key
 
 import (
 	"context"
+	"errors"
+	"net"
+	"net/url"
+	"time"
 )
 
 // Store is a key store that persists keys that
 // are referenced by a unique name.
 type Store interface {
+	// Status returns the current state of the
+	// Store.
+	Status(context.Context) (StoreState, error)
```

Comment:
Document how to interpret a returned error compared to "StoreUnreachable" state.